openedx-unsupported/ecommerce
# ENT-135: Add EnterpriseCustomer field to coupons

**Description:**

This pull request adds the ability for an ecommerce administrator to create coupon codes that are linked to a particular Enterprise Customer object.

**JIRA:** [ENT-135](https://openedx.atlassian.net/browse/ENT-135)

**Dependencies:** Depends on [ENT-112](https://github.com/edx/edx-enterprise/pull/30); in addition, this cannot be tested effectively until changes are made to that PR to enable OAuth API integration.

**Merge deadline:** 30 January 2017

**Installation instructions:**

To do manual testing, you must have edx-enterprise from the above-linked branch installed in your LMS. In addition, if the changes there haven't been made yet, you must modify the `EnterpriseCustomerViewSet` in that PR to use the `OAuth2Authentication` class, as [here](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_structure_api/v0/views.py#L32) in `edx-platform`.

**Screenshot:**

(Notice the *Enterprise Customer* field in the right-side column)
![Coupon screenshot](https://i.imgur.com/W1BYWST.png)

**Testing instructions:**

1. With the ecommerce stack and LMS running, go to http://localhost:8002/login; grant access to the ecommerce stack. You must be using a staff user or above.
2. Go to http://localhost:8002/coupons/new.
    - Note that when using a course selection mode that is not a single catalog selection, an Enterprise Customer dropdown is visible, and it shows a list of the Enterprise Customer objects installed in your LMS.
3. Create a new coupon using one of your Enterprise Customers; go to the Edit view and observe that the dropdown still shows the Enterprise Customer you chose.
    - Change the name of the EC in the LMS and refresh this page; observe that the name changes here as well.

**Reviewers:**

**Author concerns:**

I discovered while implementing this that it does not appear to be possible to edit existing coupons; based on checking against master, I do not believe this regression was introduced in this code, nor did I want to spend the time to properly diagnose and fix the issue.
